src/mesh/coordinates.cxx

```diff
@@ -275,18 +275,9 @@ Coordinates::Coordinates(Mesh* mesh, Options* options)
   g23 = interpolateAndExtrapolate(g23, location, extrapolate_x, extrapolate_y);
 
   // Check input metrics
-  if ((!finite(g11, RGN_NOBNDRY)) || (!finite(g22, RGN_NOBNDRY))
-      || (!finite(g33, RGN_NOBNDRY))) {
-    throw BoutException("\tERROR: Diagonal metrics are not finite!\n");
-  }
-  if ((min(g11, RGN_NOBNDRY) <= 0.0) || (min(g22, RGN_NOBNDRY) <= 0.0)
-      || (min(g33, RGN_NOBNDRY) <= 0.0)) {
-    throw BoutException("\tERROR: Diagonal metrics are negative!\n");
-  }
-  if ((!finite(g12, RGN_NOBNDRY)) || (!finite(g13, RGN_NOBNDRY))
-      || (!finite(g23, RGN_NOBNDRY))) {
-    throw BoutException("\tERROR: Off-diagonal metrics are not finite!\n");
-  }
+  checkFinite(g11, "g11", "RGN_NOCORNERS");
+  checkFinite(g22, "g22", "RGN_NOCORNERS");
+  checkFinite(g33, "g33", "RGN_NOCORNERS");
```

Comment:
Is it on purpose that only the diagonal metrics are checked here, and only checked for finite-ness?

Reply:
Just a brain-freeze! thanks for spotting :+1: